src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs

```diff
@@ -53,6 +53,16 @@ public class MSBuildNuGetProject : NuGetProject
         /// </summary>
         public event EventHandler<PackageEventArgs> PackageReferenceRemoved;
 
+        /// <summary>
+        /// Event to be raised when batch processing of install/ uninstall packages starts at a project level
+        /// </summary>
+        public event EventHandler<PackageEventArgs> BatchStart;
```

Comment:
Nothing in the comments nor the name ``IVsPackageInstallerProjectEvents`` conveys that. If this is only for packages.config the interface and or event names should change along with the comments.

Treating packages.config vs project.json differently is problematic with the APIs since there is no API for callers to query to find out what type of project NuGet thinks it is. If an extension wants to react to any type of NuGet operation they won't be able to use this batch API.

Also, even if these events are only for packages.config I still don't see a reason why this should be on the project itself, and the Notify method public. NuGetPackageManager is starting and ending the batch, the project is just being modified by the package manager.